File: 035_Search_Insert_Position/035_1.c
int searchInsert(int* nums, int numsSize, int target){
    if (numsSize == 0) return 0;
    int lo = 0, hi = numsSize;
    while (lo < hi) {
        int mid = (lo + hi) >> 1;
        if (target == nums[mid]) return mid;
        else if (target < nums[mid])
            hi = mid;
        else if (target > nums[mid])
            lo = mid + 1;
    }
    return lo;
}

File: 035_Search_Insert_Position/035_1.py
from typing import List

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if not nums:
            return 0

        lo, hi = 0, len(nums)
        while lo < hi:
            mid = (lo + hi) >> 1
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                lo = mid + 1
            else:
                hi = mid
        return lo